editors/vscode/src/client.ts

```diff
@@ -187,7 +187,7 @@ function nextProofPosition(document: TextDocument, proofState : Position, direct
     //Looking for the next line with "proof" in it (or the beggining/end of file)
     let line : string = document.lineAt(current).text;
 
-    while(!line.includes("proof") && current + direction >= 1 && current + direction <= document.lineCount - 1) {
+    while(!(/^\s*opaque\s+symbol/.test(line)) && current + direction >= 1 && current + direction <= document.lineCount - 1) {
```

Comment:
What do you want to test here exactly? The beginning of a proof? If so, then it should be `while(!line.includes("begin")`.

Reply:
done